core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.spec.ts

```diff
@@ -1175,9 +1176,120 @@ describe('OptimizedSsrEngine', () => {
     });
   });
 
+  describe('getRequestContext', () => {
+    let dateSpy: jest.SpyInstance;
+    let randomUUIDSpy: jest.SpyInstance;
+    const mockDate = new Date('2023-05-26');
+
+    beforeEach(() => {
+      dateSpy = jest.spyOn(global, 'Date').mockImplementation(() => mockDate);
+      randomUUIDSpy = jest
+        .spyOn(crypto, 'randomUUID')
+        .mockReturnValue('ad90db04-a501-4dc5-9b4e-2cc2ab10d49c');
+    });
+
+    afterEach(() => {
+      dateSpy.mockReset();
+      randomUUIDSpy.mockReset();
+    });
+
+    const headers: Record<string, string> = {
+      traceparent: '00-d745f6735b44e81c0ae5410cb1fc8a0c-1b527c3828976b39-01',
+    };
+    const request = {
+      originalUrl: 'test',
+      headers,
+      get: (header: string): string | string[] | null | undefined => {
+        return headers[header];
+      },
+    } as unknown as Request;
```

Comment:
Let's move `headers` and `request` to `beforeEach` block.

Why: Ideally, unit tests shouldn't share a mutable object. If the function under test by any chance mutates those a shared object (e.g. due to a bug in the function's implementation), then it might cause unwanted interference in between the separate tests, in worst case causing false success results of tests.